packages/web/app/views/patients/panes/EncounterInvoicingPane.jsx

```diff
@@ -64,106 +67,127 @@ const InvoiceContainer = styled.div`
   border-radius: 3px;
 `;
 
+const PrintButton = styled(OutlinedButton)`
+  padding: 10px 16px;
+  min-width: 100px;
+  .MuiButton-startIcon {
+    margin-right: 0;
+  }
+`;
+
 export const EncounterInvoicingPane = ({ encounter }) => {
   const { ability } = useAuth();
-  const [openInvoiceModal, setOpenInvoiceModal] = useState();
-
-  const { data: invoice } = useEncounterInvoiceQuery(encounter.id);
+  const [openInvoiceModal, setOpenInvoiceModal] = useState(false);
+  const [printModalOpen, setPrintModalOpen] = useState(false);
+  const { data: invoice, isLoading } = useEncounterInvoiceQuery(encounter.id);
   const { data: patient } = usePatientDataQuery(encounter.patientId);
 
   const handleOpenInvoiceModal = type => setOpenInvoiceModal(type);
 
   const canWriteInvoice = ability.can('write', 'Invoice');
   const canDeleteInvoice = ability.can('delete', 'Invoice');
   const cancelable = invoice && isInvoiceEditable(invoice) && canWriteInvoice;
+  const editable = invoice && isInvoiceEditable(invoice) && canWriteInvoice;
   const deletable = invoice && invoice.status !== INVOICE_STATUSES.FINALISED && canDeleteInvoice;
   const insurancePlans = invoice?.insurancePlans.map(plan => plan.name).join(', ');
 
+  if (isLoading) {
```

Comment:
Refactored to return early for loading and empty state to remove nested ternary sections